cobalt/browser/metrics/cobalt_metrics_uploader_callback.h

```diff
@@ -17,21 +17,17 @@
 
 #include <string>
 
+#include "base/callback.h"
 #include "cobalt/h5vcc/h5vcc_metric_type.h"
 
 namespace cobalt {
 namespace browser {
 namespace metrics {
 
-// An abstract class representing the callback to be invoked when the Cobalt
-// metrics payload is ready to be uploaded.
-class CobaltMetricsUploaderCallback {
- public:
-  virtual ~CobaltMetricsUploaderCallback() = default;
-
-  virtual void Run(const cobalt::h5vcc::H5vccMetricType&,
-                   const std::string&) = 0;
-};
+typedef base::RepeatingCallback<void(
```

Comment:
Do we need to make this typedef to be a single file? Seems like not necessary?

Reply:
Happy to do something different, but this type is referenced in multiple files all over. I was trying to avoid needing to bring in a different class (e.g., CobaltMetricsUploader) just for the typedef in those cases. Do you have a recommendation on what might be better?

Reply:
Instead of creating a single file just for this typedef, maybe rename this to cobalt_metric_service_utils.h or something similar? I assume there will be some other functions that can be re-used in the future and they can all be put into a single place.

Reply:
Not a bad suggestion, but if it's OK, I kind of like the current setup. If you want the CobaltMetricsUploaderCallback type, you include `cobalt_metrics_uploader_callback.h`. In general "utils" files end up being a catchall place for a lot of stuff and it can get messy really fast.